### PR TITLE
Add EdgeConnect e2e tests

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

Please include the following:

Add EdgeConnect e2e tests

## How can this be tested?

`make test/e2e/edgeconnect`

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: test/helpers/components/operator/installation.go [hunk 1]
```diff
@@ -38,20 +38,25 @@ func InstallViaHelm(releaseTag string, withCsi bool, namespace string) features.
 func UninstallViaMake(withCSI bool) features.Func {
 	return func(ctx context.Context, t *testing.T, envConfig *envconf.Config) context.Context {
 		rootDir := project.RootDir()
-		execMakeCommand(t, rootDir, "undeploy/helm", fmt.Sprintf("ENABLE_CSI=%t", withCSI))
+		err := execMakeCommand(t, rootDir, "install", fmt.Sprintf("ENABLE_CSI=%t", withCSI))
```
Comment (waodim) [inline test/helpers/components/operator/installation.go:41]: Am i missing something or why do we call `make install` here?

Comment (andriisoldatenko) [reply test/helpers/components/operator/installation.go:41]: lol, bad copy paster good catch!

Comment (andriisoldatenko) [reply test/helpers/components/operator/installation.go]: fixed in 59591ea3

**Review by waodim — DISMISSED** [review]
Tested it - works :+1:

File: test/helpers/components/operator/installation.go [hunk 2]
```diff
@@ -38,20 +38,25 @@ func InstallViaHelm(releaseTag string, withCsi bool, namespace string) features.
 func UninstallViaMake(withCSI bool) features.Func {
 	return func(ctx context.Context, t *testing.T, envConfig *envconf.Config) context.Context {
 		rootDir := project.RootDir()
-		execMakeCommand(t, rootDir, "undeploy/helm", fmt.Sprintf("ENABLE_CSI=%t", withCSI))
+		err := execMakeCommand(t, rootDir, "undeploy/helm", fmt.Sprintf("ENABLE_CSI=%t", withCSI))
+		if err != nil {
+			t.Fatal("failed to execute make command for undeploy the operator", err)
```
Comment (StefanHauth) [inline test/helpers/components/operator/installation.go:43]:
```suggestion
			t.Fatal("failed to execute make command to undeploy the operator", err)
```

File: test/helpers/components/operator/installation.go [hunk 3]
```diff
@@ -38,20 +38,25 @@ func InstallViaHelm(releaseTag string, withCsi bool, namespace string) features.
...
+func execMakeCommand(t *testing.T, rootDir, makeTarget string, envVariables ...string) error {
 	command := exec.Command("make", "-C", rootDir, makeTarget)
 	command.Env = os.Environ()
 	command.Env = append(command.Env, envVariables...)
 
-	err := command.Run()
+	output, err := command.CombinedOutput()
 	if err != nil {
-		t.Fatal("failed to install the operator via the make command", err)
+		t.Log(string(output))
```
Comment (StefanHauth) [inline test/helpers/components/operator/installation.go:56]: The old log line still would be valuable to provide some context information.

File: test/helpers/tenant/secrets.go [hunk 4]
```diff
@@ -4,10 +4,12 @@ package tenant
 
 import (
 	"context"
+	b64 "encoding/base64"
```
Comment (StefanHauth) [inline test/helpers/tenant/secrets.go:7]:
```suggestion
	"encoding/base64"
```

**Review by StefanHauth — COMMENTED** [review]
Just some nitpicks.

File: test/testdata/secrets-samples/edgeconnect-tenant.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+name: name-of-your-edge-connect
+tenantUid: abc1234
```
Comment (StefanHauth) [inline test/testdata/secrets-samples/edgeconnect-tenant.yaml:2]:
```suggestion
tenantUid: abc12345
```

Comment (andriisoldatenko) [reply test/testdata/secrets-samples/edgeconnect-tenant.yaml:2]: why?

File: test/helpers/tenant/secrets.go (hunk 4 above)
Comment (StefanHauth) [reply test/helpers/tenant/secrets.go]: Should be probably the same. Would spoil this PR a bit, though.

Comment (andriisoldatenko) [reply test/helpers/tenant/secrets.go]: fixed

File: test/helpers/components/operator/installation.go (hunk 3 above)
Comment (andriisoldatenko) [reply test/helpers/components/operator/installation.go:56]: fixed

File: test/testdata/secrets-samples/edgeconnect-tenant.yaml (hunk 5 above)
Comment (StefanHauth) [reply test/testdata/secrets-samples/edgeconnect-tenant.yaml:2]: Tenant UUIDs always have 5 digits, would make the sample a bit more consistent.

File: test/testdata/secrets-samples/edgeconnect-tenant.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+apiServer: abc1234.dev.dynatracelabs.com
```
Comment (StefanHauth) [inline test/testdata/secrets-samples/edgeconnect-tenant.yaml:3]:
```suggestion
apiServer: abc12345.dev.apps.dynatracelabs.com

**Review by StefanHauth — CHANGES_REQUESTED** [review]

File: test/testdata/secrets-samples/edgeconnect-tenant.yaml (hunk 5 above)
Comment (andriisoldatenko) [reply test/testdata/secrets-samples/edgeconnect-tenant.yaml:2]: fixed.

File: test/testdata/secrets-samples/edgeconnect-tenant.yaml (hunk 6 above)
Comment (andriisoldatenko) [reply test/testdata/secrets-samples/edgeconnect-tenant.yaml]: fixed in c9bf935b

**Review by StefanHauth — APPROVED** [review]

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**